server/sonar-server/src/test/java/org/sonar/server/computation/step/PersistSnapshotsStepTest.java

```diff
@@ -0,0 +1,504 @@
+    Component directory = DumbComponent.builder(Component.Type.DIRECTORY, 3).setUuid("CDEF").setKey("MODULE_KEY:src/main/java/dir").addChildren(file).build();
+    Component module = DumbComponent.builder(Component.Type.MODULE, 2).setUuid("BCDE").setKey("MODULE_KEY").addChildren(directory).build();
+    Component project = DumbComponent.builder(Component.Type.PROJECT, 1).setUuid("ABCD").setKey(PROJECT_KEY).addChildren(module).build();
+    treeRootHolder.setRoot(project);
+
+    dbIdsRepository.setComponentId(project, projectDto.getId());
+    dbIdsRepository.setComponentId(module, moduleDto.getId());
+    dbIdsRepository.setComponentId(directory, directoryDto.getId());
+    dbIdsRepository.setComponentId(file, fileDto.getId());
+
+    sut.execute();
+
+    assertThat(dbTester.countRowsOfTable("snapshots")).isEqualTo(4);
+
+    SnapshotDto projectSnapshot = getUnprocessedSnapshot(projectDto.getId());
+    assertThat(projectSnapshot.getComponentId()).isEqualTo(projectDto.getId());
+    assertThat(projectSnapshot.getRootProjectId()).isEqualTo(projectDto.getId());
+    assertThat(projectSnapshot.getRootId()).isNull();
+    assertThat(projectSnapshot.getParentId()).isNull();
+    assertThat(projectSnapshot.getDepth()).isEqualTo(0);
+    assertThat(projectSnapshot.getPath()).isEqualTo("");
+    assertThat(projectSnapshot.getQualifier()).isEqualTo("TRK");
+    assertThat(projectSnapshot.getScope()).isEqualTo("PRJ");
+    assertThat(projectSnapshot.getVersion()).isEqualTo("1.0");
+    assertThat(projectSnapshot.getLast()).isFalse();
+    assertThat(projectSnapshot.getStatus()).isEqualTo("U");
+    assertThat(projectSnapshot.getCreatedAt()).isEqualTo(analysisDate);
+    assertThat(projectSnapshot.getBuildDate()).isEqualTo(now);
+
+    SnapshotDto moduleSnapshot = getUnprocessedSnapshot(moduleDto.getId());
+    assertThat(moduleSnapshot.getComponentId()).isEqualTo(moduleDto.getId());
+    assertThat(moduleSnapshot.getRootProjectId()).isEqualTo(projectDto.getId());
+    assertThat(moduleSnapshot.getRootId()).isEqualTo(projectSnapshot.getId());
+    assertThat(moduleSnapshot.getParentId()).isEqualTo(projectSnapshot.getId());
+    assertThat(moduleSnapshot.getDepth()).isEqualTo(1);
+    assertThat(moduleSnapshot.getPath()).isEqualTo(projectSnapshot.getId() + ".");
+    assertThat(moduleSnapshot.getQualifier()).isEqualTo("BRC");
+    assertThat(moduleSnapshot.getScope()).isEqualTo("PRJ");
+    assertThat(moduleSnapshot.getVersion()).isEqualTo("1.1");
+    assertThat(moduleSnapshot.getLast()).isFalse();
+    assertThat(moduleSnapshot.getStatus()).isEqualTo("U");
+    assertThat(moduleSnapshot.getCreatedAt()).isEqualTo(analysisDate);
+    assertThat(moduleSnapshot.getBuildDate()).isEqualTo(now);
+
+    SnapshotDto directorySnapshot = getUnprocessedSnapshot(directoryDto.getId());
+    assertThat(directorySnapshot.getComponentId()).isEqualTo(directoryDto.getId());
```

Comment:
Unfortunately it's not possible as here I'm testing the creation of every snapshots, and every fields must be checked